### PR TITLE
Update documentation

### PR DESCRIPTION
**Pull request by schuenke**
Our documentation only included parts of our code so far.

I added the remaining submodules "Algorithms" and "Operators" and renamed/reordered the menu entries. 

supersedes #313

### CODE REVIEW

**Comment by ckolbPTB** [comment]
We are currently using MRpro, mrpro and MRPro in the documentation. Would be great to unify this. I would suggest MRpro

**Comment by ckolbPTB** [comment]
Maybe we should review and merge this PR first #313

**Comment by schuenke** [comment]
> Maybe we should review and merge this PR first #313

Sorry, I missed that one. Approved it, so that we can merge it here afterwards. 

BTW: Was there any specific reason why we chose the `pydata_sphinx_theme`?
From the style (especially the menu) I would prefer the very common Read The Docs Theme (`sphinx_rtd_theme`), with an adjusted maximum width of the main content window (e.g. 75%). This would look like this:

grafik

File: docs/source/api.rst [hunk 1]
```diff
@@ -3,8 +3,18 @@ API
 
 This is the mrpro api documentation.
```
Comment (ckolbPTB) [inline docs/source/api.rst]: MRpro

**Comment by ckolbPTB** [comment]
> BTW: Was there any specific reason why we chose the `pydata_sphinx_theme`?

I don't think so or at least I cannot remember it. I am fine with the `sphinx_rtd_theme` style.

**Review by ckolbPTB — APPROVED** [review]

**Pull request merged**